miscellaneous_scripts/extract_apt_patch_data.py

```diff
@@ -0,0 +1,172 @@
+                                     method is to add data to this list.
+    :param upgradable_packages_data_for_impacted_packages: dict[List], Dict of all the impacted source apt packages as keys and a list of their upgradable binaries as value.
+                                                                       One of the core functionalities of this method is to add data to this dict.
+    :return patch_package_list: list, the same input parameter that is modified by this function
+    :return upgradable_packages_data_for_impacted_packages: dict[list], the same input parameter that is modified by this function
+    """
+    for package in installed_packages:
+        source_package = ""
+        source_extraction_cmd_output = subprocess.run(
+            f"dpkg -s {package} | grep ^Source", shell=True, capture_output=True, text=True
+        )
+        if source_extraction_cmd_output.returncode in RETURN_CODE_OK:
+            source_package = source_extraction_cmd_output.stdout.strip().split()[1]
+
+        if (
+            any([pckg in impacted_packages for pckg in [package, source_package]])
+            and package in upgradable_packages
+        ):
+            patch_package_list.append(package)
+            ## Add package to the dict that maintains the upgradable package data
+            if package in impacted_packages:
+                if package not in upgradable_packages_data_for_impacted_packages:
+                    upgradable_packages_data_for_impacted_packages[package] = []
+                upgradable_packages_data_for_impacted_packages[package].append(package)
+            ## Add source_package to the dict that maintains the upgradable package data
+            if source_package in impacted_packages:
+                if source_package not in upgradable_packages_data_for_impacted_packages:
+                    upgradable_packages_data_for_impacted_packages[source_package] = []
+                upgradable_packages_data_for_impacted_packages[source_package].append(package)
+    return patch_package_list, upgradable_packages_data_for_impacted_packages
+
+
+def process_generative_mode_type(args):
+    """
+    This method drives the functionality for the generative mode execution. It finds all the installed and the upgradable packages.
+    Thereafter, it sends the relevant data to process_packages method to find out all the impacted packages that can be upgraded.
+    In the ends, it dumps the data at save_result_path location.
+    """
+    impacted_packages = args.impacted_packages
+    assert impacted_packages, "Impacted packages need to be passed for generative mode."
+    installed_packages = get_package_list_using_command(run_command="apt list --installed")
+    upgradable_packages = get_package_list_using_command(run_command="apt list --upgradable")
+
+    upgradable_packages_data_for_impacted_packages = {}
+    patch_package_list = []
+    patch_package_dict = {}
+
+    process_packages(
```

Comment:
1. Change the name from `process` to something specific
2. User the returned values to ensure futureproofing

Reply:
Done